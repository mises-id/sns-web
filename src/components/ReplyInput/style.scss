.reply-input-box{
  position:fixed;
  bottom: 0;
  left: 0;
  right: 0;
  width: 100vw;
  background: white;
  padding: 10px 15px;
  z-index: 1111;
}
.input{
  height: 40px;
  border-radius: 500px;
  background-color: $themeBg;
  line-height: 40px;
  padding: 0 19px;
  margin-right: 15px;
}
.button{
  width: 65px;
  height: 30px;
}