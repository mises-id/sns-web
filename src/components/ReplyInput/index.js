/*
 * @Author: lmk
 * @Date: 2021-12-06 09:34:18
 * @LastEditTime: 2022-04-12 09:42:28
 * @LastEditors: lmk
 * @Description: 
 */
import React, { forwardRef, useState } from "react";
import { useTranslation } from "react-i18next";
import { Button, Input } from "zarm";
import "./style.scss";
const ReplyInput = ({submit,content,placeholder,setselectItem,defaultItem},ref) => {
  const {t} = useTranslation();
  const [isFocus, setisFocus] = useState(false);
  return (
    <>
      <div className={`reply-input-box  ${isFocus ? 'focus-box' : ''}`}>
        <div className="m-line-top reply-input">
          <form onSubmit={submit} className="m-flex-1 m-flex ">
            <Input placeholder={placeholder||t('commentPlaceholder')} ref={ref} className="input" clearable={false} {...content} type="text" onBlur={()=>{
              setisFocus(false)
              console.log(defaultItem);
              setselectItem(defaultItem || '')
              
            }} onFocus={()=>setisFocus(true)}></Input>
            <Button
              theme="primary"
              block
              size="xs"
              onClick={submit}
              className="button"
              shape="round"
            >
              {t("send")}
            </Button>
          </form>
        </div>
      </div>
      <div className="reply-input-box-empty"></div>
    </>
  );
};

export default forwardRef(ReplyInput);
