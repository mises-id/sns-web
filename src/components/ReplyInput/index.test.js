/*
 * @Author: lmk
 * @Date: 2021-07-15 10:12:21
 * @LastEditTime: 2022-01-04 18:00:12
 * @LastEditors: lmk
 * @Description: image test
 */
import React from 'react';
import {shallow} from 'enzyme';
import ReplyInput from '.';
describe('ReplyInput', () => {
  it('should render correctly', () => {
    const component = shallow(<ReplyInput/>)
    expect(component).toMatchSnapshot();
  });
});