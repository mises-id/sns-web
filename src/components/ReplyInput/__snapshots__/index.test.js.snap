// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReplyInput should render correctly 1`] = `
<div
  className="reply-input-box  "
>
  <div
    className="m-line-top reply-input"
  >
    <form
      className="m-flex-1 m-flex "
    >
      <Input
        className="input"
        clearable={false}
        onBlur={[Function]}
        onFocus={[Function]}
        placeholder="commentPlaceholder"
        type="text"
      />
      <Button
        block={true}
        className="button"
        disabled={false}
        ghost={false}
        htmlType="button"
        loading={false}
        prefixCls="za-button"
        shadow={false}
        shape="round"
        size="xs"
        theme="primary"
      >
        send
      </Button>
    </form>
  </div>
</div>
`;
